public/js/common/react_modal.js

```diff
@@ -51,11 +51,24 @@ var Description = React.createClass({
         };
     },
 
-    componentDidMount: function(){
-        //This loads the course json
+    componentDidMount: function() {
+        this.refresh();
+    },
+
+    componentWillUpdate: function (newProps, newState) {
```

Comment:
@ryanfan @christinem we should probably be using life cycle hooks like this one more.